### ISSUE TITLE
K8s: Metac Ideas ~ Selectors, Flags, Trigger

### ISSUE THREAD

**Issue by AmitKumarDas** [AmitKumarDas/Decisions#220]
## Need for selectors:
- NameSelector
- LabelSelector
- AnnotationSelector
### ResourceRefSelector struct - set on watch or attachments
```yaml
resourceRefSelector: "spec.pvcClaimRef"
```
### NameRefSelector string - set on attachments refers to watch's name
- Is a path available in the attachment resource
```yaml
nameRefSelector: "spec.nodeName"
```
```yaml
nameRefSelector: "spec.source.persistentVolumeName"
```

### UIDRefSelector string - set on attachments refers to watch's UID
- Is a path available in the attachment resource
```yaml
uidRefSelector: "spec.nodeUID"
```
```yaml
uidRefSelector: "spec.source.persistentVolumeUID"
```

## Boolean / Flag
- IsNoop - Should this entire reconcile operation be forgot? 
  - In other words, do not proceed with further reconciliation

## Sync/Finalize Hooks as K8s Jobs
- Use of K8s Job as sync or finalize hooks
- Metac will aggregate the current watch & attachments state & pass them as configmaps to K8s Job

## Sync/Finalize Hooks as K8s Deployments
- Binary uses Metac as a library
- Binary uses a config (_controller specifications_) via ConfigMap
- Binary runs a service which in turn
  - Provides the config to Metac lib
  - Implements the sync function exposed by Metac
